### PR TITLE
[Rust] Use lance local object store

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/src/dataset.rs [hunk 1]
```diff
@@ -822,14 +826,16 @@ impl Dataset {
...
     indices: Option<Vec<Index>>,
 ) -> Result<()> {
     let paths = vec![
-        manifest_path(object_store.base_path(), manifest.version),
-        latest_manifest_path(object_store.base_path()),
+        manifest_path(base_path, manifest.version),
+        latest_manifest_path(base_path),
     ];
 
+    println!("Writting manifest file to {:?}", paths);
```
Comment (wjones127) [inline rust/src/dataset.rs:838]: still need this?

File: rust/src/dataset.rs [hunk 2]
```diff
@@ -1384,6 +1390,7 @@ mod tests {
     #[tokio::test]
     async fn test_open_dataset_not_found() {
         let result = Dataset::open(".").await;
+        println!("{:?}", result);
```
Comment (wjones127) [inline rust/src/dataset.rs:1393]: and this?

File: rust/src/io/object_store.rs [hunk 3]
```diff
@@ -168,8 +174,11 @@ impl ObjectStore {
         &self.base_path
     }
 
-    /// Open a file for path
+    /// Open a file for path.
+    ///
+    /// The [path] is absolute path.
     pub async fn open(&self, path: &Path) -> Result<Box<dyn ObjectReader>> {
+        println!("open path: {:?} schema: {}", path, self.scheme);
```
Comment (wjones127) [inline rust/src/io/object_store.rs:181]:
```suggestion
```

**Review by wjones127 — APPROVED** [review]
Seems reasonable. It seems to allow spaces and some other characters like `:` in paths. Emojis aren't allowed in paths though, but IMO that's fine.

File: rust/src/dataset.rs (hunk 1 above)
Comment (eddyxu) [reply rust/src/dataset.rs:838]: fixed

File: rust/src/dataset.rs (hunk 2 above)
Comment (eddyxu) [reply rust/src/dataset.rs:1393]: fixed.

File: rust/src/io/object_store.rs (hunk 3 above)
Comment (eddyxu) [reply rust/src/io/object_store.rs]: fixed

**Review by chebbyChefNEQ — APPROVED** [review]

**Pull request merged**